usaspending_api/etl/elasticsearch_loader_helpers/controller.py

```diff
@@ -115,31 +122,34 @@ def run_deletes(self):
         client = instantiate_elasticsearch_client()
         if self.config["data_type"] == "award":
             deleted_awards(client, self.config)
-        else:
+        elif self.config["data_type"] == "transaction":
             deleted_transactions(client, self.config)
+        else:
+            raise RuntimeError(f"No delete function implemented for type {self.config['data_type']}")
 
 
-def extract_transform_load(worker):
+def extract_transform_load(task):
     if abort.is_set():
-        logger.info(format_log(f"{worker.name} became lost on the way over."))
+        logger.info(format_log(f"{task.name} partition was skipped due to previous error"))
         return
 
     start = perf_counter()
-    a = choice(["skillfully", "deftly", "expertly", "readily", "quickly", "nimbly", "casually", "easily", "boldly"])
-    logger.info(format_log(f"{worker.name.upper()} {a} enters the arena", job=worker.name))
+    msg = f"Started processing on partition #{task.partition_number}: {task.name}"
+    logger.info(format_log(msg, name=task.name))
 
     client = instantiate_elasticsearch_client()
     try:
-        records = worker.transform_func(worker, extract_records(worker))
-        load_data(worker, records, client)
+        records = task.transform_func(task, extract_records(task))
+        if abort.is_set():
+            logger.info(format_log(f"Prematurely ending {task.name} due to previous error"))
+            return
+        load_data(task, records, client)
     except Exception:
         if abort.is_set():
-            logger.info(format_log(f"{worker.name} realizes the battle lost. #ragequit"))
+            logger.info(format_log(f"{task.name} failed after an error was previously encountered"))
         else:
-            msg = f"{worker.name} has fallen in battle! How could such a thing happen to this great warrior?"
-            logger.error(msg, job=worker.name)
+            logger.error(format_log(f"{task.name} failed!", name=task.name))
             abort.set()
     else:
-        attrib = choice(["pro", "champ", "boss", "top dog", "hero", "super"])
-        msg = f"{worker.name} completed the mission like a {attrib} in {perf_counter() - start:.2f}s"
-        logger.info(format_log(msg, job=worker.name))
+        msg = f"Partition {task.name} was successfully processed in {perf_counter() - start:.2f}s"
```

Comment:
Also be helpful to see partition number in this log, since hopefully this would indicate how close to all 10,000 partitions you are getting.

Reply:
I have not observed the partitions initiated sequentially. Obviously they are run in parallel, but if there are 10 subprocesses I haven't observed the first 10 partitions to be 0 - 9.

Reply:
update: Testing on a smaller dataset did show the partitions 0-9 as the first 10 to run (albeit not in that order)